inspirehep/modules/authors/utils.py

```diff
@@ -126,3 +137,79 @@ def phonetic_blocks(full_names, phonetic_algorithm='nysiis'):
     )
 
     return dict(zip(full_names, phonetic_blocks))
+
+
+def formdata_to_model(formdata, id_workflow, id_user, is_update=False):
+    """Generate the record model dict and extra_data dict from the form data.
+
+    Returns:
+        tuple(dict, dict): tuple containing the dict representaion of a record
+            and the extra_data to create a new workflow.
+    """
+    form_fields = copy.deepcopy(formdata)
+    extra_data = {}
+
+    filter_empty_elements(
+        form_fields,
+        ['institution_history', 'advisors',
+         'websites', 'experiments']
+    )
+    data = updateform.do(form_fields)
+
+    data['_collections'] = ['Authors']
+
+    if '$schema' in data and not data['$schema'].startswith('http'):
+        data['$schema'] = url_for(
+            'invenio_jsonschemas.get_schema',
+            schema_path="records/{0}".format(data['$schema'])
+        )
+
+    author_name = ''
+
+    if 'family_name' in form_fields and form_fields['family_name']:
+        author_name = form_fields['family_name'].strip() + ', '
+    if 'given_names' in form_fields and form_fields['given_names']:
+        author_name += form_fields['given_names']
+
+    if author_name:
+        data.get('name', {})['value'] = author_name
+
+    if 'extra_comments' in form_fields and form_fields['extra_comments']:
+        data.setdefault('_private_notes', []).append({
+            'source': 'submitter',
+            'value': form_fields['extra_comments']
+        })
+
+    data['stub'] = False
+
+    # ==========
+    # Submitter Info
+    # ==========
+    try:
+        user_email = User.query.get(id_user).email
+    except AttributeError:
+        user_email = ''
+    try:
+        orcid = UserIdentity.query.filter_by(
+            id_user=id_user,
+            method='orcid'
+        ).one().id
+    except NoResultFound:
+        orcid = ''
+    data['acquisition_source'] = dict(
+        email=user_email,
+        datetime=datetime.datetime.utcnow().isoformat(),
+        method="submitter",
+        orcid=orcid,
+        submission_number=str(id_workflow),
+        internal_uid=int(id_user),
+    )
+
+    data = strip_empty_values(data)
+
+    validate(data, 'authors')
+
+    if is_update is not None:
```

Comment:
Maybe this can also be done later in the workflow for consistency reasons using [this](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/tasks/actions.py#L66). Maybe doing something like [this](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/workflows/article.py#L436) and then [this](https://github.com/inspirehep/inspire-next/blob/master/inspirehep/modules/workflows/workflows/article.py#L160) .

Reply:
You are 100% right, that should be a mark in the workflow. though I think it's out of the scope of this PR to refactor it.

We should tackle this when doing the refactor of the author workflow.

(btw. fyi this PR is rebased on top of #3081, this one only adds the cleanup at the end).